edge-modules/DirectMethodReceiver/src/Program.cs

```diff
@@ -37,45 +29,27 @@ static async Task<int> MainAsync()
                 .Build();
 
             TransportType transportType = configuration.GetValue("ClientTransportType", TransportType.Amqp_Tcp_Only);
-            Console.WriteLine($"Using transport {transportType.ToString()}");
+            ModuleClient moduleClient = await ModuleUtil.CreateModuleClientAsync(
+                transportType,
+                Logger,
+                ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                ModuleUtil.DefaultTransientRetryStrategy);
 
-            await InitModuleClient(transportType);
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), null);
 
-            // Wait until the app unloads or is cancelled
-            var cts = new CancellationTokenSource();
-            AssemblyLoadContext.Default.Unloading += (ctx) => cts.Cancel();
-            Console.CancelKeyPress += (sender, cpe) => cts.Cancel();
-            await WhenCancelled(cts.Token);
-            return 0;
-        }
+            await moduleClient.OpenAsync();
+            await moduleClient.SetMethodHandlerAsync("HelloWorldMethod", HelloWorldMethodAsync, null, cts.Token);
+            await cts.Token.WhenCanceled();
 
-        static async Task InitModuleClient(TransportType transportType)
-        {
-            ITransportSettings[] GetTransportSettings()
-            {
-                switch (transportType)
-                {
-                    case TransportType.Mqtt:
-                    case TransportType.Mqtt_Tcp_Only:
-                    case TransportType.Mqtt_WebSocket_Only:
-                        return new ITransportSettings[] { new MqttTransportSettings(transportType) };
-                    default:
-                        return new ITransportSettings[] { new AmqpTransportSettings(transportType) };
-                }
-            }
-
-            ITransportSettings[] settings = GetTransportSettings();
-
-            ModuleClient moduleClient = await ModuleClient.CreateFromEnvironmentAsync(settings).ConfigureAwait(false);
-            await moduleClient.OpenAsync().ConfigureAwait(false);
-            await moduleClient.SetMethodHandlerAsync("HelloWorldMethod", HelloWorldMethod, null).ConfigureAwait(false);
-
-            Console.WriteLine("Successfully initialized module client.");
+            completed.Set();
+            handler.ForEach(h => GC.KeepAlive(h));
+            Logger.LogInformation("DirectMethodReceiver Main() finished.");
+            return 0;
         }
 
-        static Task<MethodResponse> HelloWorldMethod(MethodRequest methodRequest, object userContext)
+        static Task<MethodResponse> HelloWorldMethodAsync(MethodRequest methodRequest, object userContext)
```

Comment:
Nit: It is fine that you have already done it, but no need to changes this to Async (here any everywhere)